File: project/.sdk/src/cmp/lua/fragment/EntityListOp.fragment.lua
local entityListOp = {}

-- EJECT-START

function EntityNameEntity:list(reqmatch, ctrl)
  local utility = self._utility
  local ctx = utility.make_context({
    opname = "list",
    ctrl = ctrl,
    match = self._match,
    data = self._data,
    reqmatch = reqmatch,
  }, self._entctx)

  return self:_run_op(ctx, function()
    if ctx.result ~= nil then
      if ctx.result.resmatch ~= nil then
        self._match = ctx.result.resmatch
      end
    end
  end)
end

-- EJECT-END

File: project/.sdk/src/cmp/py/fragment/EntityListOp.fragment.py
entity_list_op = None

# EJECT-START

    def list(self, reqmatch, ctrl=None):
        utility = self._utility
        ctx = utility.make_context({
            "opname": "list",
            "ctrl": ctrl,
            "match": self._match,
            "data": self._data,
            "reqmatch": reqmatch,
        }, self._entctx)

        def post_done():
            if ctx.result is not None:
                if ctx.result.resmatch is not None:
                    self._match = ctx.result.resmatch

        return self._run_op(ctx, post_done)

# EJECT-END
